File: examples/AppModel/booksandrecords-spring-boot/src/main/scala/com/ms/booksandrecords/commands/CloseDeal.scala
package com.ms.booksandrecords.commands

import com.fasterxml.jackson.annotation.JsonProperty
import org.springframework.stereotype.Component

@Component class CloseDeal() extends DealCmd {
  var id = ""

  def this(@JsonProperty("id") id: String) {
    this()
    this.id = id
  }

  def getId = id

  def setId(id: String) = this.id = id
}

File: examples/AppModel/booksandrecords-spring-boot/src/main/java/com/ms/booksandrecords/commands/CloseDeal.java
package com.ms.booksandrecords.commands;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.springframework.stereotype.Component;

@Component
public class CloseDeal extends DealCmd {
    String id;
    public CloseDeal(){}
    public CloseDeal(@JsonProperty("id") String id) {

        this.id = id;
    }
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }
}
